### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Matt1994**
What we believe to be the MVP

- Users can create a card and document as a guest
- Logged in users have the document fields saved to a personal profile
- This personal profile can be accessed by scanning the QR code on the back of the card

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ var passport = require('passport');
 var LocalStrategy = require('passport-local').Strategy;
 var mongo = require('mongodb');
```
Comment (mavagio) [inline app.js:11]: where are LocalStrategy and mongo used? are they global variables ? in any case its a bad practice.

File: app.js [hunk 2]
```diff
@@ -19,6 +20,8 @@ var routes = require('./routes/index');
 var users = require('./routes/users');
```
Comment (mavagio) [inline app.js:20]: Same with db, never used.

File: public/css/style.css [hunk 3]
```diff
@@ -7,7 +7,7 @@ html {
 }
```
Comment (mavagio) [inline public/css/style.css:7]: on the line 3, @import prevents parallel downloads. Using link is recommended.

File: public/js/medid/creator.js [hunk 4]
```diff
@@ -1,66 +1,225 @@
...
 			$(this).parent().parent().remove();
 		});
+
+    field.find('.moveUp').on('click', function() {
+			row = $(this).parent().parent();
+      row.prev().before(row);
+		});
+
+    field.find('.moveDown').on('click', function() {
+      row = $(this).parent().parent();;
```
Comment (mavagio) [inline public/js/medid/creator.js:127]: double semicolon ;

File: public/js/medid/creator.js [hunk 5]
```diff
@@ -72,33 +231,65 @@ define(['jquery'], function($) {
...
-  /* This method must be called by the document-specific
-   * controller when it is initialized itself.
-   * This method checks for any user data.
+
+
+  /**
+   * This method must be called by the document-specific controller when it is initialized itself.
+   * This method checks for any user data and puts it into the pre-set form.
    */
   creator.init = function () {
```
Comment (mavagio) [inline public/js/medid/creator.js:270]: The function has multiple levels of abstraction, I point to here but there are other functions in this module that have the same problem. Remember, it is recommended that each function does one thing.

File: routes/users.js [hunk 6]
```diff
@@ -34,30 +34,38 @@ router.post('/register', function(req, res){
 	var errors = req.validationErrors();
```
Comment (mavagio) [inline routes/users.js:34]: line 24: where is password2 used, what is its purpose in this module?

**Review by mavagio — APPROVED** [review]
Remarks about the code can be found on Basecamp. Here are some minor comments.

File: routes/users.js (hunk 6 above)
Comment (Matt1994) [reply routes/users.js:34]: It is used on Line 32, it essentially works as the 'confirm' password, as seen on other sites, to ensure the user has typed it in correctly the first time

**Pull request merged**